src/chat/tools/base.ts

```diff
@@ -63,3 +60,25 @@ export abstract class BaseLanguageModelTool<T> implements LanguageModelTool<T> {
     } as LanguageModelChatTool;
   }
 }
+
+/**
```

Comment:
I think this comment should make it to the eventual GH discussion, so the approach is laid out for other devs to see